### ISSUE TITLE
Restricted filesystem permissions are not shown in permission inspector

### ISSUE THREAD

**Issue by lordloh** [sjarvela/kloudspeaker#105]
![capture](https://cloud.githubusercontent.com/assets/2434923/12609099/a2872a3e-c494-11e5-9a5c-3d43df2552aa.PNG)
I have given the above permissions to a user on a folder, but the upload buttons do not appear on the folder page. The exact same permission to a different user gets an upload button on the UI. What could be going wrong?

**Comment by sjarvela** [comment]
You should use permission inspector to see what is the effective permission.

With admin, right click the desired item (or use folder dropdown menu in header) and choose "Permissions...". There click tab "User permission", and choose the correct user to inspect.

It will display what is the effective permission, and the list below will show all permissions there are, in the  order of significance (topmost is effective).

**Comment by lordloh** [comment]
I am the admin and see "User is administrator and has all permissions.", however the upload icon does not appear in the folder. Have I got some permissions wrong?

The upload icon also does not appear for users with permission - "Read&Write, No delete"

The user's effective permissions are - 

| User | Permission | Source |
| --- | --- | --- |
| group1 | Read&Write, No delete | This |
| userName | Read Only | User default |
| group2 | Read&Write, No delete | Group default |
|  | Read Only | System default |

**Comment by sjarvela** [comment]
I'm having hard to follow this, you are talking about different users but say "I am admin and see...". Which user are you inspecting, and what are the effective permissions to which users? Admins should have all permissions, and there is no use to inspect him.

It sounds like you are talking about two different users, both with same setup? Both nonadmins?

And what is the exact setup
- what groups do the users have, both have same?
- what permissions do the users have
- what permissions do the groups have
- there are no permissions to the folder itself?
- both users have the same folder assigned? I mean there are not two folders listed in admin/Folders, each for both users, but instead only one folder which is assigned to both users?
- user default is readonly, correct?
- system default is readonly, correct?

Is it possible to have temporary admin account for the site so I can see it myself? This is a bit vague issue, hard to solve with so little info

**Comment by lordloh** [comment]
Where may I send you the credentials for an admin and a user to test?

**Comment by lordloh** [comment]
It looks like I was able to iron out a lot of things by checking the linux filesystem permissions. I had to add apache to my group and grant group write permission to folders that I shared out of my home directory. In addition, I have some of these folders shared over the local network using samba. This is adding further permission issues - I need to fix these. 

I was worried about the permissions, as I would not see the "New Folder" and "Upload" buttons, even after logging in as admin (because of linux file system permissions). Would it be possible to include linux file system permissions while showing effective permissions? or at least warn the admin that file system permissions will not permit kloudspeaker to operate as desired? [Feature Request?]

**Comment by sjarvela** [comment]
Okay, that makes sense. There was a pull request some time ago to support restricted filesystem permissions, ie. kloudspeaker will identify the case where the system does not allow writing. The use case was a bit marginal, it does not make much sense to restrict the writability in filesystem level if you intend to give write permissions in Kloudspeaker.

I did mention in that pull request that it won't show up in permission inspector, but I agree it should.
